### ISSUE TITLE
VST3

### ISSUE THREAD

**Issue by giohappy** [helgoboss/realearn#885]
@helgoboss thanks for this great plugin! Any plan to support more recent plugin formats like VLST3 or CLAP?

**Comment by helgoboss** [comment]
Hi! Both are supported already. Did you find something specific that doesn't work with VST3 or CLAP plug-ins?

**Comment by giohappy** [comment]
Great. Where can I download the vst3 and clap versions from? The release apparently includes only the vst2 DLL (I'm on Windows). Reaper only shows VSTi and no VST3i or CLAPi plugin.

**Comment by helgoboss** [comment]
Ah, you mean ReaLearn itself as a VST3/CLAP? No. Why? It wouldn't make any difference in functionality. ReaLearn is not the average instrument/effect plug-in where this would matter. It's totally different and can integrate into REAPER in ways that are far beyond the possibilities of CLAP/VST3 ... simply because it uses the REAPER API directly (instead of indirectly via plug-in standards).

It's possible that I will change to CLAP in future but not for functional reasons, only for licensing reasons that could affect me as a developer but not users.

**Comment by giohappy** [comment]
Yes, sure, I mean for licensing not for functional reasons. I would like to test some changes and possible contributions but I don't own a VST 2 license and therefore I cannot build the plugin. 

Ok, maybe my first contribution could be in that direction :)

**Comment by helgoboss** [comment]
You can still build it of course, I guess just publishing is a grey zone. There's only a tiny portion of code in ReaLearn that is actually tied to the VST plug-in standard. The problem of porting to another standard is mainly a problem of making that change convenient to the user. E.g. making it possible to load an older project (which used VST version) with automatic migration to the latest CLAP version of ReaLearn. Ideally, this change should be almost transparent for users.

If you intent to contribute, be aware that ReaLearn will undergo quite some structural changes soon with the advent of Playtime 2 and until this is done, I will not have the time to review PRs. And even then, if you have plans for bigger contributions to ReaLearn, please let's discuss them in detail before you actually go ahead and implement the PR. I received some PRs in the past (reaper-rs but also ReaLearn) that I couldn't accept because they were touching just too many things at once or changed existing API/behavior or I didn't have the capacity to look into them in a timely manner. Would be good to prevent that sort of thing in future.

**Comment by helgoboss** [comment]
vst-rs for sure doesn't ship the official SDK files. It just exports some functions and communicates with the host in certain ways that (by coincidence or not) resemble what an official VST2 plug-in would do. Whether this is legal or not, I have no idea. I believe it's a grey zone. If I'm correct, that's also why vst-rs is archived nowadays.

**Comment by giohappy** [comment]
>vst-rs for sure doesn't ship the official SDK files. It just exports some functions and communicates with the host in certain ways that (by coincidence or not) resemble what an official VST2 plug-in would do. Whether this is legal or not, I have no idea. I believe it's a grey zone. If I'm correct, that's also why vst-rs is archived nowadays.

Oh, I really thought it was binding the SDK. 
vst-rs is for sure another good reason to switch to another protocol. [clap-sys](https://github.com/glowcoil/clap-sys) might be a good candiate.
